### ISSUE TITLE
[Source Request]: Stadt Kleve

### ISSUE THREAD

**Issue by Rama-Drama** [mampfes/hacs_waste_collection_schedule#3234]
### Municipality / Region

Kleve

### Collection Calendar Webpage

https://buerger-app-klevestadt.azurewebsites.net/calendar/

### Example Address

Stadionstraße 2, 47533 Kleve

### Collection Data Format

Dedicated IOS/Android APP

### Additional Information

Die Stadt Kleve nutzt auch die App c-trade die ich hier schon finden könnte unter den Quellen, nur die Konfiguration wird nicht angezeigt.

**Comment by ss83** [comment]
Looking for the same integration.
Beside the mobile app one can also export the ics-file from the website.

**Comment by ss83** [comment]
I looked for the traffic of the webpage and it goes to:

https://apps.c-trace.de/klevestadt-buergerapp/....

Assuming "klevestadt-buergerapp" is the service. But it seems that it's not going to http://web.c-trace.de/...

**Comment by ss83** [comment]
looking for the same service provider.

The webapp uses the service "klevestadt-buergerapp". 

Here is a snippet:
https://apps.c-trace.de/klevestadt-buergerapp/OData.svc/AbfuhrtermineAbJahr?$expand=Abfuhrplan%2CAbfuhrplan%2FAbfallartenObj&$orderby=Abfuhrplan%2FAbfallartenObj%2FName%2CBemerkung&jahr=2025&ort=%27Kleve%27&strasse=%27FLIERSOL%27&hausNr=%2721%27&$format=json

**Comment by ss83** [comment]
Could it be that the service provider is "buergerportal_de".
By comparing, it seems more to be similar to the portal.

**Comment by Rama-Drama** [comment]
Also der Anbieter der Lösung ist C-Trace. 
Bisher habe ich das immer nur mit der CSV Datei gemacht, welches auch super klappt. Aber ich will es nicht jedes Jahr manuell machen müssen. 

Hier habe ich noch ne Auflistung der Felder gefunden:
https://apps.c-trace.de/klevestadt-buergerapp/OData.svc/

**Comment by ss83** [comment]
Das Bürger Portal scheint auch im Hintergrund von C-Trace zu sein.

Welche zugangsdaten hast du da verwendet bei dem link?

**Comment by ss83** [comment]
Any progress here? I would be willing to help, if something is missing :-)

**Comment by waveFrontSet** [comment]
I don't think it's viable to extend the `c_trace_de` or the `buergerportal_de` sources to integrate Kleve.
1. `c_trace_de` seems to rely on a session id that is fetched from a response header.
2. `buergerportal_de` seems to rely on an API url.
The analogous URLs for Kleve lead to 401s or 404s, respectively. In fact, the frontend app uses basic authentication to communicate with the backend (I don't want to expose username / password here, but you may use your browser's developer tools to inspect the requests and obtain the Base64-encoded username + password from the `Authorization` header).

To sum it up, I think an entirely new source is needed to integrate Kleve, unfortunately. 🙁

**Comment by nl2rma** [comment]
Hi all, I have added a pull request that adds support for Umweltbetriebe USK Kleve.

**Comment by ss83** [comment]
Thank you very much.
Just tested yesterday, but unfortunately I only get results for "Glas" "Gelbe Tonne" "Wertstoffe nach tel. Anmeldung" "Schadstoffmobil siehe Entsorgungsorte".

What is missing are the normal "Restmüll" "Papier" "Bio".

It is for Kleve, Fliersol e.g. Also tested different streets.

**Comment by nl2rma** [comment]
Hi @ss83  I have two instances of HAOS running, test and prod, and on both setups I have all the available.

![Image](https://github.com/user-attachments/assets/66270a1b-d68b-4051-8636-8eaec85dafff)

To be sure I looked up the street FLIERSOL 5 in de the online schedule and I see exactly what you described:
https://buerger-app-klevestadt.azurewebsites.net/calendar
![Image](https://github.com/user-attachments/assets/925173e1-3c2c-494f-8e75-d7e07135ff5a)

So noting is wrong with the code, try for example LINDENALLEE 24 and that the online schedule looks exactly like mine calendar in HAOS
![Image](https://github.com/user-attachments/assets/9cbe3265-996b-4959-8bbe-19a02471721c)

Try switching to the next week view in the calendar and than you will see more and different entries for sure.

**Comment by ss83** [comment]
Thanks. Just checken on the homepage. There all is Okay by now.  So I removed Fliersol and added it again. Now all is shown correctly. Maybe it was an fault on buergerportal. Removed and added it several times this morning with no change. But now it works :-)

Thank you for the work!

**Comment by nl2rma** [comment]
No problem, fine to hear. I was looking into this, at least for my self because I am going to move to Kleve in Juli.Greetings.Op 6 mei 2025 22:37 schreef Sven ***@***.***>:ss83 left a comment (mampfes/hacs_waste_collection_schedule#3234)
Thanks. Just checken on the homepage. There all is Okay by now.  So I removed Fliersol and added it again. Now all is shown correctly. Maybe it was an fault on buergerportal. Removed and added it several times this morning with no change. But now it works :-)
Thank you for the work!

**Comment by 5ila5** [comment]
added in #4088 

This is be part of the 2.8.0 release
